logstash-core/lib/logstash/pipeline.rb

```diff
@@ -323,61 +393,70 @@ def flush_filters(options = {}, &block)
     end
   end
 
+  class FlusherObserver
+    def initialize(logger, flushing)
+      @logger = logger
+      @flushing = flushing
+    end
+
+    def update(time, result, exception)
```

Comment:
I disagree. I read the code and do not understand quickly the purpose so something is missing to help me make the mental link between this and concurrent ruby and timer task and what not. 

It might be obvious for you but you have to think about other people coming in the code and having to figure what this is about and what it does. Here it is not obvious to understand the purpose of the `FlusherObserver` class. 

As I said, maybe a few lines of comments would help both for the class and method.